test/plugins/trigger.test.js

```diff
@@ -876,147 +876,1466 @@ describe('trigger tests', () => {
+        const downstreamPipeline2 = await pipelineFactoryMock.createFromFile('~sd@1:a-downstream.yaml');
 
-        await eventFactoryMock.create({
-            pipelineId: parentPipeline.id,
+        const upstreamEvent = await eventFactoryMock.create({
+            pipelineId: upstreamPipeline.id,
             startFrom: 'hub'
         });
 
         // run all builds
-        await buildFactoryMock.run();
+        await upstreamEvent.run();
 
-        const childEvent1 = childPipeline1.getLatestEvent();
-        const childEvent2 = childPipeline2.getLatestEvent();
+        const downstreamEvent1 = downstreamPipeline1.getLatestEvent();
+        const downstreamEvent2 = downstreamPipeline2.getLatestEvent();
 
-        assert.equal(childEvent1.getBuildOf('target').status, 'SUCCESS');
-        assert.equal(childEvent2.getBuildOf('target').status, 'SUCCESS');
+        assert.equal(downstreamEvent1.getBuildOf('target').status, 'RUNNING');
+        assert.equal(downstreamEvent2.getBuildOf('target').status, 'RUNNING');
     });
 
-    it('[ ~sd@2:a ] is triggered in a upstream', async () => {
-        const parentPipeline = await pipelineFactoryMock.createFromFile('~sd@2:a-parent.yaml');
+    it('[ ~sd@1:a ] is not triggered in a downstream when sd@1:a fails', async () => {
+        const upstreamPipeline = await pipelineFactoryMock.createFromFile('~sd@1:a-upstream.yaml');
+        const downstreamPipeline = await pipelineFactoryMock.createFromFile('~sd@1:a-downstream.yaml');
 
-        await pipelineFactoryMock.createFromFile('~sd@2:a-child.yaml');
+        const upstreamEvent = await eventFactoryMock.create({
+            pipelineId: upstreamPipeline.id,
+            startFrom: 'hub'
+        });
 
-        const event = await eventFactoryMock.create({
-            pipelineId: parentPipeline.id,
+        await upstreamEvent.getBuildOf('hub').complete('SUCCESS');
+        await upstreamEvent.getBuildOf('a').complete('FAILURE');
+
+        assert.isNull(downstreamPipeline.getLatestEvent());
+        assert.equal(downstreamPipeline.getBuildsOf('target').length, 0);
+    });
+
+    it('[ ~sd@2:a ] is not triggered in a upstream when sd@1:a fails', async () => {
+        const upstreamPipeline = await pipelineFactoryMock.createFromFile('~sd@2:a-upstream.yaml');
+        const downstreamPipeline = await pipelineFactoryMock.createFromFile('~sd@2:a-downstream.yaml');
+
+        const upstreamEvent = await eventFactoryMock.create({
+            pipelineId: upstreamPipeline.id,
             startFrom: 'hub'
         });
 
-        // run all builds
-        await buildFactoryMock.run();
+        await upstreamEvent.run();
 
-        assert.equal(event.getBuildOf('target').status, 'SUCCESS');
+        const downstreamEvent = downstreamPipeline.getLatestEvent();
+
+        await downstreamEvent.getBuildOf('a').complete('FAILURE');
+
+        assert.isNull(upstreamEvent.getBuildOf('target'));
```

Comment:
We also want to make sure that the build is not running on a new event.
```suggestion
        assert.equal(upstreamPipeline.getBuildsOf('target').length, 0);
```